src/ssl.c

```diff
@@ -47166,14 +47243,25 @@ static int wolfSSL_TicketKeyCb(WOLFSSL* ssl,
 
     (void)ctx;
 
-    if (ticketKeyCb == NULL)
+    WOLFSSL_ENTER("wolfSSL_TicketKeyCb");
+
+    if (ssl == NULL || ssl->ctx == NULL || ssl->ctx->ticketEncCtx == NULL) {
+        WOLFSSL_MSG("Bad parameter");
         return WOLFSSL_TICKET_RET_FATAL;
+    }
 
-    wolfSSL_EVP_CIPHER_CTX_init(&evpCtx);
     /* Initialize the cipher and HMAC. */
-    res = ticketKeyCb(ssl, keyName, iv, &evpCtx, &hmacCtx, enc);
-    if (res != TICKET_KEY_CB_RET_OK && res != TICKET_KEY_CB_RET_RENEW)
+    wolfSSL_EVP_CIPHER_CTX_init(&evpCtx);
```

Comment:
I don't see a need this anymore.

Reply:
The user supplies the key and IV into this struct so that we can encrypt the ticket with it,